pkg/projects/resource_project_user.go

```diff
@@ -0,0 +1,214 @@
+package projects
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type User struct {
+	Name  string   `json:"name"`
+	Roles []string `json:"roles"`
+}
+
+type Comparable interface {
+	Hash() string
+	Equals(other Comparable) bool
+}
+
+func (user User) Hash() string {
+	return user.Name
+}
+
+func (a User) Equals(b User) bool {
+	return a.Hash() == b.Hash()
+}
+
+func contains(as []User, b User) bool {
+	log.Printf("[DEBUG] contains")
+	log.Printf("[TRACE] as: %+v\n", as)
+	log.Printf("[TRACE] b: %+v\n", b)
+
+	for _, a := range as {
+		log.Printf("[TRACE] a: %+v\n", a)
+		log.Printf("[TRACE] a.Equals(b): %+v\n", a.Equals(b))
+		if a.Equals(b) {
+			return true
+		}
+	}
+	return false
+}
+
+// Not the most efficient way to determine the slices intersection but this suffices for the small-ish number of items
+var usersIntersection = func(as []User, bs []User) []User {
+	log.Printf("[DEBUG] usersIntersection")
+	log.Printf("[TRACE] as: %+v\n", as)
+	log.Printf("[TRACE] bs: %+v\n", bs)
+
+	intersection := make([]User, 0)
+
+	for _, a := range as {
+		if contains(bs, a) {
+			intersection = append(intersection, a)
+		}
+	}
+
+	return intersection
+}
+
+// Not the most efficient way to determine the slices differences but this suffices for the small-ish number of items
```

Comment:
What if you did something a little more sophisticated:

```go
func userApply(predicate func (user a,b) bool) func (users []User) []User {
 
  return func(users []User) []User {
// do iterate code here but use predicate for your `if`
   }
}

var userDiff = userApply(func (a, bs User) bool {
  return !contains(bs,a)
}
diff := userDiff(users1,users2)

This will eliminate the copy-pasta